tests/devices/test_default_qubit.py

```diff
@@ -2077,7 +2100,7 @@ class TestApplyOps:
     """Tests for special methods listed in _apply_ops that use array manipulation tricks to apply
     gates in DefaultQubit."""
 
-    state = np.arange(2**4).reshape((2, 2, 2, 2))
+    state = np.arange(2**4, dtype=np.complex256).reshape((2, 2, 2, 2))
```

Comment:
Would it be worth asserting the device dtype in the test cases below too?

Reply:
I think it is unnecessary as tests in this class just check whether the algorithms we used for operations are correct. I still may add another class that checks each operation preserves dtypes.

Reply:
But this must be `complex128`.